common/resource/resource_impl.go

```diff
@@ -253,6 +256,13 @@ func New(
 		serviceConfig.IsErrorRetryableFunction,
 	)
 
+	shardDistributorRawClient := clientBean.GetShardDistributorClient()
+	shardDistributorClient := retryable.NewShardDistributorClient(
```

Comment:
nit: might be checked for nil? If the client is nil, there is no point in creating a retryable client.

Reply:
Thanks, thats a good point, fixed!